src/pages/index.js

```diff
@@ -94,32 +168,78 @@ function closeModal(modal) {
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModal);
+  const data = {
+    name: editModalNameInput.value,
+    about: editModalDescriptionInput.value,
+  };
+
+  setButtonText(editSubmitBtn, true);
+
+  api
+    .editUserInfo(data)
+    .then((updatedUser) => {
+      profileName.textContent = updatedUser.name;
+      profileDescription.textContent = updatedUser.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(editSubmitBtn, false);
+    });
 }
 
 function handleNewPostFormSubmit(evt) {
   evt.preventDefault();
+
   const inputValues = {
     name: editCardNameInput.value,
     link: editCardLinkInput.value,
   };
 
-  evt.target.reset();
+  setButtonText(cardSubmitBtn, true);
+
+  api
+    .addCard(inputValues)
+    .then(() => {
+      return api.getInitialCards();
```

Comment:
Please, delete `api.getInitialCards();` from the card form submit handler because you should `prepend` a card which comes from the server response. It's already there


```js
  setButtonText(cardSubmitBtn, true);

  api
    .addCard(inputValues)
    .then((latestCard) => {  // the latest card comes here
      const cardElement = getCardElement(latestCard);
      cardsList.prepend(cardElement);
      closeModal(cardModal);

```